### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Ooblioob**
@m3brown Can you do a quick review of this before we merge it all in?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,98 +1,78 @@
...
-
-1. Projects that inspired you
-2. Related projects
-3. Books, papers, talks, or other sources that have meaniginful impact or influence on this project
+jenkins-automation
+==================
+
+Repos automated by Jenkins DSL 
+
+This repository contains the automation scripts for dev Jenkins.
```
Comment (m3brown) [inline README.md]: pointing out 'dev Jenkins' seems a little off for the first sentence of a public readme..

File: README.md [hunk 2]
```diff
@@ -1,98 +1,78 @@
...
+
+### Jenkins Plugins
+
+- https://wiki.jenkins-ci.org/display/JENKINS/Job+DSL+Plugin
+- https://wiki.jenkins-ci.org/display/JENKINS/Build+Flow+Plugin
+- https://wiki.jenkins-ci.org/display/JENKINS/EnvInject+Plugin
+- http://wiki.jenkins-ci.org/display/JENKINS/Git+Plugin
+- http://wiki.jenkins-ci.org/display/JENKINS/Multiple+SCMs+Plugin
+- http://wiki.jenkins-ci.org/display/JENKINS/NodeJS+Plugin
+- http://wiki.jenkins-ci.org/display/JENKINS/Clone+Workspace+SCM+Plugin
```
Comment (m3brown) [inline README.md:35]: Are all these plugins required? recommended? minimum/suggested versions?

File: README.md [hunk 3]
```diff
@@ -1,98 +1,78 @@
...
+
+### Job DSL Bootstrap
```
Comment (m3brown) [inline README.md]: I went cross-eyed reading the Bootstrap procedure.  I think the biggest reason is you're talking about bootstrapping to create stuff but I don't know what the stuff is.  What is bootstrap creating?  Also, how do I copy this sample code into a jenkins job?

File: jenkins-automation.groovy [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+job('template-base') {
+  wrappers {
+    colorizeOutput()
+  }
+
+  // handle failure
+}
+
+
+job('template-base-js') {
```
Comment (m3brown) [inline jenkins-automation.groovy:13]: Is the intent to have templates for all of our stacks here? `template-base-python`, `template-base-qu`, etc.?  Won't that make this file kind of messy/big? Is it possible to break this up and import?  Also, will this work cleanly if we have two different version of python or will we need `template-base-python26` and `template-base-python27`?

File: jenkins-automation.groovy [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+// ** DSL Master Job **
+// ********************
+projectRepos = readFileFromWorkspace("project_repos.txt").split('\n')
+
+job('DSL Master') {
+  using 'template-base'
+  multiscm {
+    git {
+      remote {
+        url('https://github.com/dgreisen-cfpb/jenkins-automation')
```
Comment (m3brown) [inline jenkins-automation.groovy]: Shouldn't this be cfpb isntead of `dgreisen-cfpb`?

File: lib/loadRepositories.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+
+def load_repositories(repo_name, is_pinned, workspace_dir):
+    repo_data, proj_name = build_repo_data(repo_name, is_pinned, workspace_dir)
+    
+    for pinned_url in repo_data['repositories']:
+        clone_repo(pinned_url, is_pinned, workspace_dir, proj_name+'-repos')
+
+    try:
+        unitybox_repository = repo_data['unitybox_repository']
```
Comment (m3brown) [inline lib/loadRepositories.py:84]: I don't see the requirement for unitybox detailed anywhere.  what is it and why is it a requirement?

File: README.md (hunk 2 above)
Comment (Ooblioob) [inline README.md:35]: Addressed, we're calling these "Required" now.

File: jenkins-automation.groovy (hunk 4 above)
Comment (Ooblioob) [inline jenkins-automation.groovy:13]: Noted for later, this is a great suggestion!

File: lib/loadRepositories.py (hunk 6 above)
Comment (Ooblioob) [inline lib/loadRepositories.py:84]: README is still in progress, but we'll add this in when we get closer to finishing.

**Comment by m3brown** [comment]
Looks good.

**Pull request merged**